File: app/src/main/java/com/MohamedTaha/Imagine/New/ui/fragments/GeneralFragment.java
package com.MohamedTaha.Imagine.New.ui.fragments;

import android.os.Bundle;

import androidx.preference.PreferenceFragmentCompat;

import com.MohamedTaha.Imagine.New.R;

public class GeneralFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.general_preferences, rootKey);
    }
}

File: app/src/main/java/com/mohamedtaha/imagine/ui/fragments/GeneralFragment.kt
package com.mohamedtaha.imagine.ui.fragments

import androidx.preference.PreferenceFragmentCompat
import android.os.Bundle
import com.mohamedtaha.imagine.R
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class GeneralFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.general_preferences, rootKey)
    }
}